pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -115,7 +115,7 @@ func (r *TokenREST) Create(ctx context.Context, name string, obj runtime.Object,
 		}
 	}
 	if len(out.Spec.Audiences) == 0 {
-		out.Spec.Audiences = r.auds
+		out.Spec.Audiences = r.audsSet.List()
```

Comment:
The `List` call performs a sort under the hood. We could make the `audsSet` and `auds` two separate fields, but i feel the duplication could be prone to errors. I briefly considered creating a `NewTokenRequest` function storage package that could avoid duplication for callers. Irrespective there will be two similar looking fields on the struct.

Or We could make the `List` call performant by sorting the `auds` before creating the `audsSet` .

Which one do you prefer?

Reply:
@liggitt lmk.

Reply:
yeah, leave this bit alone and create the second field. `NewREST` is the only place we construct `TokenREST`, so that's a reasonable place to construct `audsSet`